core/src/main/scala/kafka/metrics/clientmetrics/ClientMetricsReceiverPlugin.scala

```diff
@@ -17,22 +17,26 @@
 
 package kafka.metrics.clientmetrics
 
-import org.apache.kafka.common.metrics.{ClientTelemetryPayload, ClientTelemetryReceiver}
-import org.apache.kafka.common.requests.PushTelemetryRequest
+import org.apache.kafka.common.metrics.ClientTelemetryReceiver
+import org.apache.kafka.common.requests.{PushTelemetryRequest, RequestContext}
+
+import scala.collection.mutable.ListBuffer
 
 object ClientMetricsReceiverPlugin {
-  var cmReceiver :Option[ClientTelemetryReceiver] = None
-  def getCmReceiver() = cmReceiver
+  val cmReceivers  = new ListBuffer[ClientTelemetryReceiver]()
+  def isEmpty = cmReceivers.isEmpty
+  def reset = cmReceivers.empty
 
-  def init(receiver: ClientTelemetryReceiver) = {
-    cmReceiver = Option(receiver)
+  def add(receiver: ClientTelemetryReceiver) = {
+    cmReceivers.append(receiver)
   }
 
-  def createPayload(request: PushTelemetryRequest): ClientTelemetryPayload = {
-    new ClientMetricsPayload(request.getClientInstanceId,
-                             request.isClientTerminating,
-                             request.getMetricsContentType,
-                             request.getMetricsData)
+  def exportMetrics(context: RequestContext, request: PushTelemetryRequest): Unit = {
+    val payload = new ClientMetricsPayload(request.getClientInstanceId,
+                                           request.isClientTerminating,
+                                           request.getMetricsContentType,
+                                           request.getMetricsData)
```

Comment:
Not sure how paranoid to be here -- the byte array is being passed into each plugin. One plugin could alter the contents, right?

Reply:
I guess to be on safe side let's make a copy for each plugin.

Reply:
Is it possible to only copy it if there's a need? That is, only copy it on the 2..n plugins, so that if there's only one plugin we can just use the byte buffer as is.

Reply:
@kirktrue how about just returning read only buffer, looks like ByteBuffer has the interface to return asReadOnlyBuffer.. then plugin can make a copy if it needs to modify it.

Reply:
Great! As long as that also prevents accessing the internal byte array, that sounds like a good way to go.